sycl/include/CL/sycl/stl.hpp

```diff
@@ -72,5 +72,48 @@ unique_ptr_class<T> make_unique_ptr(ArgsT &&... Args) {
   return unique_ptr_class<T>(new T(std::forward<ArgsT>(Args)...));
 }
 
+template <typename T = void> using plus = std::plus<T>;
+template <typename T = void> using multiplies = std::multiplies<T>;
+template <typename T = void> using bit_or = std::bit_or<T>;
+template <typename T = void> using bit_xor = std::bit_xor<T>;
+template <typename T = void> using bit_and = std::bit_and<T>;
+template <typename T = void> using logical_and = std::logical_and<T>;
+template <typename T = void> using logical_or = std::logical_or<T>;
+
+template <typename T = void> struct minimum {
+  T operator()(const T &lhs, const T &rhs) const {
+    return std::less<T>()(lhs, rhs) ? lhs : rhs;
+  }
+};
+
+template <> struct minimum<void> {
+  struct is_transparent {};
```

Comment:
This one is not in the SYCL 2020 spec.

Reply:
[Section 4.17.2](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:function-objects) says:

> All function objects obey C++ conversion and promotion rules. Each function object is additionally specialized for `void` as a transparent function object that deduces its parameter types and return type.

The intent of this paragraph is to say that these function objects should behave like the ones [defined in standard C++](http://eel.is/c++draft/function.objects#arithmetic.operations.plus), which specialize for `void` and add an `is_transparent` member.  Do you think this part of the specification needs to be clarified?

Reply:
Ah, that makes sense, I missed this while skimming over the spec. I think it's clear enough.